File: java-advent-of-code/src/test/java/advent/twenty_seventeen/Day16PermutationPromenadeTest.java
package advent.twenty_seventeen;

import advent.common.DailyProblem;
import org.junit.Test;

import static org.junit.Assert.*;

public class Day16PermutationPromenadeTest {
    @Test
    public void checkReal() {
        DailyProblem<String, String> dp = new Day16PermutationPromenade("/twenty_seventeen/Day16-PermutationPromenade-input.txt", 16);

        assertEquals("bijankplfgmeodhc", dp.getPart1Answer());
        assertEquals("bpjahknliomefdgc", dp.getPart2Answer());
    }

    @Test
    public void checkExample1() {
        DailyProblem<String, String> dp = new Day16PermutationPromenade("/twenty_seventeen/Day16-PermutationPromenade-example#1.txt", 5);

        assertEquals("baedc", dp.getPart1Answer());
        assertEquals("abcde", dp.getPart2Answer());
    }
}

File: scala-advent-of-code/src/test/scala/advent/twenty_seventeen/Day16PermutationPromenadeTest.scala
package advent.twenty_seventeen

import org.scalatest.{FlatSpec, Matchers}

class Day16PermutationPromenadeTest  extends FlatSpec with Matchers {
  "2017 Day 16 - Input" should "calculate answers" in {
    val dp = new Day16PermutationPromenade("/twenty_seventeen/Day16-PermutationPromenade-input.txt")
    dp.part1Answer should be(0)
    dp.part2Answer should be(0)
  }

  "2015 Day 9 - Example #1" should "calculate answers" in {
    val dp = new Day16PermutationPromenade("/twenty_seventeen/Day16-PermutationPromenade-example#1.txt")
    dp.part1Answer should be(0)
    dp.part2Answer should be(0)
  }
}
